SingularityUI/app/components/newDeployForm/NewDeployForm.jsx

```diff
@@ -0,0 +1,1303 @@
+          if (!this.hasValue(volume[fieldId])) {
+            return false;
+          }
+        }
+      }
+    }
+    if (this.props.request.request.loadBalanced) {
+      for (const fieldId of REQUIRED_FIELDS.loadBalancer) {
+        if (!this.hasValue(this.getValueOrDefault(fieldId))) {
+          return false;
+        }
+      }
+    }
+    return true;
+  }
+
+  addFieldsToDeployObject(deployObject, fieldsToAdd) {
+    for (const fieldId of fieldsToAdd) {
+      if (fieldId.type === 'object') {
+        deployObject[fieldId.id] = this.addFieldsToDeployObject({}, fieldId.values);
+      } else if (this.hasValue(this.getValueOrDefault(fieldId.id))) {
+        deployObject[fieldId.id] = this.getValueOrDefault(fieldId.id);
+      }
+    }
+    return deployObject;
+  }
+
+  submit(event) {
+    event.preventDefault();
+    const deployObject = {};
+    this.addFieldsToDeployObject(deployObject, FIELDS.all);
+    if (this.getValueOrDefault('executorType') === DEFAULT_EXECUTOR_TYPE) {
+      this.addFieldsToDeployObject(deployObject, FIELDS.defaultExecutor);
+    } else {
+      this.addFieldsToDeployObject(deployObject, FIELDS.customExecutor);
+    }
+    deployObject.requestId = this.props.request.request.id;
+    this.props.save({deploy: deployObject});
+  }
+
+  addThingToArrayField(fieldId, thing) {
+    if (!this.getValue(fieldId)) {
+      this.updateField(fieldId, [thing]);
+    } else {
+      const fieldValue = this.getValue(fieldId).slice();
+      fieldValue.push(thing);
+      this.updateField(fieldId, fieldValue);
+    }
+  }
+
+  addThingPreventDefault(fieldId, thing, event) {
+    event.preventDefault();
+    this.addThingToArrayField(fieldId, thing);
+  }
+
+  removeThingFromArrayField(fieldId, key) {
+    const fieldValue = this.getValue(fieldId).slice();
+    fieldValue.splice(key, 1);
+    this.updateField(fieldId, fieldValue);
+  }
+
+  updateThingInArrayField(fieldId, key, newFieldObj) {
+    const newArray = this.getValue(fieldId).slice();
+    const newValue = _.extend({}, newArray[key], newFieldObj);
+    newArray[key] = newValue;
+    this.updateField(fieldId, newArray);
+  }
+
+  renderDefaultExecutorFields() {
+    const cmdLineArguments = (
+      <div className="form-group">
+        <label htmlFor="cmd-line-args">Arguments</label>
+        <MultiInput
+          id = "cmd-line-args"
+          value = {this.getValue('arguments') || []}
+          onChange = {(newValue) => this.updateField('arguments', newValue)}
+        />
+      </div>
+    );
+    const artifacts = (
+      <div className="form-group">
+        <label htmlFor="artifacts" >Artifacts</label>
+        <MultiInput
+          id = "artifacts"
+          value = {this.getValue('uris') || []}
+          onChange = {(newValue) => this.updateField('uris', newValue)}
+          placeholder="eg: http://s3.example/my-artifact"
+        />
+      </div>
+    );
+    return (
+      <div>
+        <fieldset id="default-expandable" className='expandable'>
```

Comment:
Run this through `eslint` (not `gulp lint`, the plugin is broken for inherited configuration for some odd reason). The className should be double quotes.